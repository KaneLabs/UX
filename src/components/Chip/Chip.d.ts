export default Chip;
declare function Chip({ text, flat, style, hoverStyle, focusStyle, children, ...rest }: {
    [x: string]: any;
    text?: any;
    flat?: boolean | undefined;
    style?: any;
    hoverStyle?: any;
    focusStyle?: any;
    children: any;
}): JSX.Element;
