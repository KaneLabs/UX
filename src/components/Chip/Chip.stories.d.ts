declare const _default: import("@storybook/addons").StoryApi<import("@storybook/react/dist/client/preview/types").StoryFnReactReturnType>;
export default _default;
