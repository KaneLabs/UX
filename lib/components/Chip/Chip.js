var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _slicedToArray2=_interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));var _objectWithoutProperties2=_interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));var _react=_interopRequireWildcard(require("react"));var _reactNative=require("react-native");var _theme=require("../../theme");var _Paper=_interopRequireDefault(require("../Paper"));var _Typography=_interopRequireDefault(require("../Typography"));var _this=this,_jsxFileName="/Users/kane/Lab/Eros/src/components/Chip/Chip.tsx";var DEFAULT_CHIP_HEIGHT=32;var Chip=function Chip(props){var text=props.text,_props$flat=props.flat,flat=_props$flat===void 0?false:_props$flat,style=props.style,hoverStyle=props.hoverStyle,focusStyle=props.focusStyle,children=props.children,rest=(0,_objectWithoutProperties2.default)(props,["text","flat","style","hoverStyle","focusStyle","children"]);var _useState=(0,_react.useState)(false),_useState2=(0,_slicedToArray2.default)(_useState,2),focus=_useState2[0],setFocus=_useState2[1];var _useState3=(0,_react.useState)(false),_useState4=(0,_slicedToArray2.default)(_useState3,2),hover=_useState4[0],setHover=_useState4[1];var styles=useStyles();var ChipStyle=_react.default.useMemo(function(){return _reactNative.StyleSheet.flatten([styles.Chip,hover&&styles.ChipHovered,hover&&hoverStyle,focus&&styles.ChipFocused,focus&&focusStyle,flat&&styles.ChipFlat,_reactNative.Platform.OS==='web'&&{outlineWidth:0},style]);},[]);return _react.default.createElement(_Paper.default,(0,_extends2.default)({style:ChipStyle,onFocus:function onFocus(){return setFocus(true);},onBlur:function onBlur(){return setFocus(false);},onMouseEnter:function onMouseEnter(){return setHover(true);},onMouseLeave:function onMouseLeave(){return setHover(false);}},rest,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:50,columnNumber:5}}),text&&_react.default.createElement(_Typography.default,{type:_theme.TypographyTypes.body2,__self:_this,__source:{fileName:_jsxFileName,lineNumber:57,columnNumber:17}},text)||children||null);};var useStyles=(0,_theme.makeStyles)(function(theme){return{Chip:_reactNative.Platform.select({default:{height:DEFAULT_CHIP_HEIGHT,borderRadius:DEFAULT_CHIP_HEIGHT,borderColor:theme.borderColor,borderWidth:theme.borderWidth,paddingLeft:theme.padding,paddingRight:theme.padding,overflow:'hidden',justifyContent:'center'}}),ChipFlat:_reactNative.Platform.select({web:{backgroundColor:theme.canvas2Opaque,borderWidth:0,outlineWidth:0},default:{backgroundColor:theme.canvas2Opaque,borderWidth:0}}),ChipHovered:(0,_extends2.default)({borderColor:theme.textColor.disabled,color:theme.textColor.secondary},theme.shadow(8),{backgroundColor:theme.canvas2Opaque}),ChipFocused:(0,_extends2.default)({borderColor:theme.textColor.secondary,color:theme.textColor.primary},theme.shadow(8),{backgroundColor:theme.canvas2Opaque})};});var _default=Chip;exports.default=_default;