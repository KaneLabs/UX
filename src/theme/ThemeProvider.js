import React, {
  useState, useMemo, createContext, useCallback,
} from 'react';
import { useColorScheme } from 'react-native';
import * as light from './light';
import * as dark from './dark';

export const ThemeContext = createContext(null);

export const ThemeProvider = (props) => {
  const { children, initialScheme } = props;
  const colorScheme = useColorScheme();
  console.log(`Detected device color scheme: ${colorScheme}`);
  const [themeMode, setThemeMode] = useState(initialScheme || colorScheme);
  const toggleTheme = useCallback(() => setThemeMode((mode) => (mode === 'light' ? 'dark' : 'light')));
  const value = useMemo(
    () => (themeMode === 'light' ? [light, toggleTheme] : [dark, toggleTheme]),
    [themeMode],
  );

  return (
    <ThemeContext.Provider value={value}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;
