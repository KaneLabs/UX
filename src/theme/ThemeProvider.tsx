import React, {
  useState,
  useMemo,
  createContext,
  useCallback,
  ReactNode,
} from 'react';
import { useColorScheme } from 'react-native';
import * as light from './light';
import * as dark from './dark';

export const ThemeContext = createContext(null);

interface ThemeProviderProps {
  initialScheme?: string;
  children: ReactNode;
}

interface ThemeType {}

export type ThemeContextType = unknown | [ThemeType, Function];

export type ThemeProvider<ThemeProviderProps> = (
  props: ThemeProviderProps,
) => React.ContextType<any>;

export const ThemeProvider = (props: ThemeProviderProps) => {
  const { children, initialScheme } = props;
  const colorScheme = useColorScheme();
  console.log(
    `Detected device color scheme: ${colorScheme}`,
    `initial color scheme: ${initialScheme || colorScheme}`,
  );

  const [themeMode, setThemeMode] = useState(initialScheme || colorScheme);
  const toggleTheme = useCallback(
    () => setThemeMode(mode => (mode === 'light' ? 'dark' : 'light')),
    [themeMode],
  );
  const value = useMemo(
    () => (themeMode === 'light' ? [light, toggleTheme] : [dark, toggleTheme]),
    [themeMode],
  );

  return (
    <ThemeContext.Provider value={value}>{children}</ThemeContext.Provider>
  );
};

export default ThemeProvider;
