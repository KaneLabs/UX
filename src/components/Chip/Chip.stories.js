import React from 'react';
import { View } from 'react-native';

import Chip from './Chip';

import Container from 'eros-ui-storybook/components/Container';
import { storiesOf } from 'eros-ui-storybook/helpers/storiesOf';

export default storiesOf('Components|Chip', module)
  .addParameters({ jest: ['Chip'] })
  .add('Chip', () => (
    <Container>
      <View style={{ flexDirection: 'column', flexWrap: 'wrap' }}>
        <Chip text="Default" />
      </View>
    </Container>
  ));
