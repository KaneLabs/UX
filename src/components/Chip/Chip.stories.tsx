import React from 'react';
import { View } from 'react-native';

import Chip from './Chip';

import Container from '@kanelabs/ux-storybook/components/Container';
import { storiesOf } from '@kanelabs/ux-storybook/helpers/storiesOf';

export default storiesOf('Components|Chip', module)
  .addParameters({ jest: ['Chip'] })
  .add('Chip', () => (
    <Container>
      <View style={{ flexDirection: 'column', flexWrap: 'wrap' }}>
        <Chip text="Default" />
      </View>
    </Container>
  ));
