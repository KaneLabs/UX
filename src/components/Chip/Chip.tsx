import React, { useState } from 'react';
import { View, Platform } from 'react-native';

import { useTheme, makeStyles, shadow } from 'eros-ui/theme';
import Paper from 'eros-ui/components/Paper';
import Typography from 'eros-ui/components/Typography';

const DEFAULT_CHIP_HEIGHT = 32;

const useStyles = makeStyles(theme => ({
  default: Platform.select({
    default: {
      height: DEFAULT_CHIP_HEIGHT,
      borderRadius: DEFAULT_CHIP_HEIGHT,
      // maxWidth: DEFAULT_CHIP_HEIGHT * 3,
      borderColor: theme.borderColor,
      borderWidth: theme.borderWidth,
      paddingLeft: theme.padding,
      paddingRight: theme.padding,
      overflow: 'hidden',
      justifyContent: 'center',
    },
  }),
  flat: Platform.select({
    web: {
      backgroundColor: theme.canvas2Opaque,
      borderWidth: 0,
      outlineWidth: 0,
    },
    default: { backgroundColor: theme.canvas2Opaque, borderWidth: 0 },
  }),
}));

const Chip = ({
  text = null,
  flat = false,
  style = null,
  hoverStyle = null,
  focusStyle = null,
  children,
  ...rest
}) => {
  const [{ textColor, canvas2Opaque }] = useTheme();
  const [focus, setFocus] = useState(false);
  const [hover, setHover] = useState(false);
  const styles = useStyles();

  const ChipStyle = [
    styles.default,
    hover && {
      borderColor: textColor.disabled,
      color: textColor.secondary,
      ...shadow(8),
      backgroundColor: canvas2Opaque,
      ...hoverStyle,
    },
    // focus && {
    //   borderColor: textColor.secondary,
    //   color: textColor.primary,
    //   ...shadow(8),
    //   backgroundColor: canvas2Opaque,
    //   ...focusStyle,
    // },
    flat && styles.flat,
    Platform.OS === 'web' && { outlineWidth: 0 }, // needed for Web
    style,
  ];

  return (
    <Paper
      style={ChipStyle}
      onFocus={() => setFocus(true)}
      onBlur={() => setFocus(false)}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
      {...rest}>
      {(text && <Typography type="body2">{text}</Typography>) ||
        children ||
        null}
    </Paper>
  );
};

export default Chip;
