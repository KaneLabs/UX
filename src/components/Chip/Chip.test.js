import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Chip from './Chip';
import ThemeProvider from '../../theme/ThemeProvider';

const renderer = new ShallowRenderer();

test('Chip Renders', () => {
  const tree = renderer.render(
    <ThemeProvider>
      <Chip text="Default" />
    </ThemeProvider>,
  );
  expect(tree).toMatchSnapshot();
});
