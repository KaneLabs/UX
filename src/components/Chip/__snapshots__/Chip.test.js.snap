// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Renders 1`] = `
<Context.Provider
  value={
    Array [
      Object {
        "ASIDE_WIDTH": 300,
        "AVATAR_SIZE": 40,
        "CHAT_WIDTH": 192,
        "DEFAULT_AVATAR_URL": "https://art.fora.co/default-avatar.png",
        "DESKTOP_BREAKPOINT": 900,
        "DRAWER_WIDTH": 192,
        "FEED_WIDTH": 600,
        "FEED_WIDTH_DENSE": 420,
        "NAV_HEIGHT": 48,
        "PROFILE_AVATAR_SIZE": 90,
        "TABBAR_HEIGHT": 56,
        "TOOLBAR_WIDTH_CLOSED": 64,
        "TOOLBAR_WIDTH_OPEN": 192,
        "Typography": Object {
          "body1": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 16,
            "fontWeight": "300",
            "letterSpacing": 0.5,
            "lineHeight": 19,
          },
          "body2": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 14,
            "fontWeight": "300",
            "letterSpacing": 0.25,
            "lineHeight": 17,
          },
          "button": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 1.25,
            "lineHeight": 18,
          },
          "caption": Object {
            "color": "rgba(0,0,0,0.54)",
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 12,
          },
          "h1": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 96,
            "fontWeight": "200",
            "letterSpacing": -1.5,
            "lineHeight": 116,
          },
          "h2": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 60,
            "fontWeight": "200",
            "letterSpacing": -0.5,
            "lineHeight": 72,
          },
          "h3": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 40,
            "fontWeight": "300",
            "letterSpacing": 0,
            "lineHeight": 48,
          },
          "h4": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 28,
            "fontWeight": "300",
            "letterSpacing": 0.25,
            "lineHeight": 34,
          },
          "h5": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 24,
            "fontWeight": "300",
            "letterSpacing": 0,
            "lineHeight": 29,
          },
          "h6": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          "overline": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 10,
            "fontWeight": "300",
            "letterSpacing": 1.5,
            "lineHeight": 12,
          },
          "subtitle1": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 16,
            "fontWeight": "300",
            "letterSpacing": 0.15,
            "lineHeight": 20,
          },
          "subtitle2": Object {
            "color": "rgba(0,0,0,0.87)",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 17,
          },
        },
        "appBackgroundColor": "rgba(240,240,240,1)",
        "background": "rgba(240,240,240,1)",
        "backgroundColor": "rgba(240,240,240,1)",
        "backgroundColorOpaque": "rgba(240,240,240,0.67)",
        "backgroundColorUltraOpaque": "rgba(240,240,240,0.05)",
        "borderColor": "rgba(0,0,0,0.12)",
        "borderColorHover": "rgba(0,0,0,0.87)",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "canvas": "rgba(255,255,255, 1)",
        "canvas1": "rgba(255,255,255, 1)",
        "canvas1Opaque": "rgba(255,255,255,0.67)",
        "canvas2": "rgba(255,255,255, 1)",
        "canvas2Opaque": "rgba(255,255,255,0.67)",
        "canvas3": "rgba(255,255,255, 1)",
        "canvas3Opaque": "rgba(255,255,255,0.67)",
        "canvasOpaque": "rgba(255,255,255,0.67)",
        "gutter": 16,
        "iconColor": "rgba(0,0,0,0.54)",
        "iconPadding": 12,
        "iconSize": 24,
        "mode": "light",
        "padding": 12,
        "shadow": [Function],
        "textColor": Object {
          "disabled": "rgba(0,0,0,0.38)",
          "faint": "rgba(0,0,0,0.12)",
          "hint": "rgba(0,0,0,0.38)",
          "inactive": "rgba(0,0,0,0.38)",
          "primary": "rgba(0,0,0,0.87)",
          "secondary": "rgba(0,0,0,0.54)",
        },
        "unit": 8,
      },
      [Function],
    ]
  }
>
  <Chip
    text="Default"
  />
</Context.Provider>
`;
